### ISSUE TITLE
weird ` ,SV[string]` in help msg

### ISSUE THREAD

**Issue by timotheecour** [c-blake/cligen#54]
```
toast -h
Usage:
  main [required&optional-params]
  Options(opt-arg sep :|=|spc):
  -h, --help                                  write this help to stdout
  -m=, --mode=         string       "cpp"     language; see CompileMode
  -p, --past           bool         false     print AST output
  --pnim               bool         false     run preprocessor on header
  --pretty             bool         true      set pretty
  --preprocess         bool         false     print Nim output
  -d=, --defines=      ,SV[string]  EMPTY     set defines
  -i=, --includeDirs=  ,SV[string]  EMPTY     include directory to pass to preprocessor
  -s=, --source=       string       REQUIRED  C/C++ source/header
```

* why the `,` in `,SV[string] ` ?
* what does SV stand for? (it's not in README)

**Comment by c-blake** [comment]
",SV" is a more international-friendly "CSV" - i.e., comma separated values, but it's rare/my invention.  It's also more general.  If the CLI author changes the delimiter to ":" then that should read ":SV", etc.  CSV/TSV/DSV all seemed too vague.  I'm kind of hoping "<literal-delimiter>SV" catches on.

**Comment by c-blake** [comment]
(Closed because this is already documented pretty extensively in the `argcvt` module docs.  And if you hate it instead of seeing its charm you can, in theory, override all of the arg parsers by just defining your own before `dispatchGen` gets called.  It's not hard to cut & paste and hack up what I have defined, either.  So, this is at most an argument about how parameterized various default behaviors should be.)

**Comment by c-blake** [comment]
Oh, and hey - *at least* I didn't default to DPSV by default.  ;-)

**Comment by c-blake** [comment]
(Oh, and defining your own `argAggHelp` to say whatever you want can probably be like 4 lines of code.  The more general one is only 8.)

**Comment by timotheecour** [comment]
I've added test cases in D20181130T174608 and D20181130T172327 in https://github.com/timotheecour/vitanim/blob/master/testcases/tests/t0000.nim for additional issues I'm seeing with this.

the default value formatting for seq params is not good:
* newlines are not escaped, unlike for non-seq string params (see D20181130T172327)
  * `@[""]` prints as nothing_at_all (for DPSV, <D>)
  * `@[]` prints as EMPTY (for DPSV, EMPTY, which is not super consistent)
* it's neither DRY (DPSV or ,SV is repeated in every seq entry) nor standard notation
* it's ambiguous, eg when delimiter = " " (a space), default value for `@["", ""]` is indistinguishable from default value for `@[""]` (admittedly, this is rare scenario but still)

test case:
```nim
when defined(case11):
  proc main(foo1: int, foo2: float, foo3: seq[string] = @["baz1", "baz2"], foo4: seq[string] = @[], foo5: seq[string] = @[""], foo6: seq[int] = @[], foo7: seq[int] = @[10], foo8="")=
    discard
```

without DPSV:
```
Usage:
  main [required&optional-params]
  Options(opt-arg sep :|=|spc):
  -h, --help                            write this help to stdout
  -f=, --foo1=  int          REQUIRED   set foo1
  --foo2=       float        REQUIRED   set foo2
  --foo3=       ,SV[string]  baz1,baz2  set foo3
  --foo4=       ,SV[string]  EMPTY      set foo4
  --foo5=       ,SV[string]             set foo5
  --foo6=       ,SV[int]     EMPTY      set foo6
  --foo7=       ,SV[int]     10         set foo7
  --foo8=       string       ""         set foo8
```

with DPSV:
```
Usage:
  main [required&optional-params]
  Options(opt-arg sep :|=|spc):
  -h, --help                                  write this help to stdout
  -f=, --foo1=  int           REQUIRED        set foo1
  --foo2=       float         REQUIRED        set foo2
  --foo3=       DPSV[string]  <D>baz1<D>baz2  set foo3
  --foo4=       DPSV[string]  EMPTY           set foo4
  --foo5=       DPSV[string]  <D>             set foo5
  --foo6=       DPSV[int]     EMPTY           set foo6
  --foo7=       DPSV[int]     <D>10           set foo7
  --foo8=       string        ""              set foo8
```


## proposal: same notation regardless of delimiter used; plus 1 extra line show which delimiter is used (when relevant)
```
Usage:
  main [required&optional-params]
  Options(opt-arg sep :|=|spc):
  -h, --help                            write this help to stdout
  -f=, --foo1=  int               REQUIRED          set foo1
  --foo2=       float             REQUIRED          set foo2
  --foo3=       seq[string]  ["baz1", "baz2"]  set foo3
  --foo4=       seq[string]  []                          set foo4
  --foo5=       seq[string]  [""]                       set foo5
  --foo5b=     seq[string] ["", ""] .                 set foo5
  --foo6=       seq[int]     []                            set foo6
  --foo7=       seq[int]     [10]                         set foo7
  --foo8=       string       ""                             set foo8
```

**Comment by c-blake** [comment]
It seems to me that what should be in the help message should be about what CLI users are supposed to type.  So, that's why the default values are "spelled" the way a CLI user would enter them, not with Nim types/notation as you are suggesting.  The CLI user may not be a Nim programmer or have any idea what a `seq` is.  It can be hard to "perspective take" when one is juggling all the different roles - CLI meta-package author, API author, CLI author, and CLI user, but that help messages are for CLI users is uncontentious.

What I like about `,SV` or `:SV` is that it _is_ a consistent notation regardless of delimiter, generalized from CSV, TSV, DSV that also works "inline" in the few text columns we want for a terse summary.  The point about a space not rendering is valid.  I special case `TAB` in `TSV` due to unprintability concerns.  I know `,SV` is not _standard_, but the standard is not consistent.

DPSV is just a totally different thing to solve the "some delimiter that works with what I need to delimit in some specific case" problem.  Dates back to at least the earliest `sed`/`ed` impls in the 1970s.  It's quite useful at side-stepping the hassle of quoting.  I don't know of a standard name for it.  If you have a better suggestion than the name "DPSV" or my notation "<D>A<D>B", I am open-minded about that.  It seems like "<x>" is the most common "user substitutes something here", but that's not always consistent, and it's definitely not clear that it is only a single char.

We can't explain everything in a terse table, though, and existing practice isn't quite standardized for full functionality, especially for "aggregate values" to option arguments or things like `+=`.  E.g., it wouldn't be _crazy_ to demand CLI users _had_ to use `+=` for aggregates and offload quoting to the shell and have no delimiting syntax at all.  I only tested it that way when I wrote it for you which is how that `+=seq[string]` bug persisted.

I feel what might make you happy here is something else I have thought about -- having some kind of `"--help-syntax"` option or maybe a `helpSyntax` dispatch parameter with a CLI author overridable default.  That could include some kind of one- to few-line line syntax summary after the option table.  My macro code mostly offloads real type analysis to Nim proper/the overloading resolver.  So, I don't know if it would be easy to auto-detect that seq/set/etc. aggregate types were "in play somewhere" to only emit help on that when it makes sense.  Emit too much help and important stuff is lost in the noise.
